caproto/_dbr.py

```diff
@@ -1072,7 +1098,9 @@ def promote_type(ftype, *, use_status=False, use_time=False, use_ctrl=False):
         ftype += ChType.TIME_STRING
     elif use_status:
         ftype += ChType.STS_STRING
+    elif use_gr:
```

Comment:
This code block should get some comment about the assumption that the types are in blocks, all in the same order, and than the string type is the first one in each block.